t/app/model/defecttype.t

```diff
@@ -59,6 +68,44 @@ subtest 'Problem->defect_types behaves correctly' => sub {
     is $problem->defect_types->first->name, $general_defect_type->name, 'Correct defect type is returned for Traffic lights category';
 };
 
+subtest 'by_categories returns all defect types grouped by category' => sub {
+    FixMyStreet::App->model('DB::BodyArea')->create( { body => $oxfordshire, area_id => $area_id } );
+
+    my @contacts = FixMyStreet::DB->resultset('Contact')->not_deleted->search( { body_id => [ $oxfordshire->id ] } )->all;
+    my $defect_types = FixMyStreet::App->model('DB::DefectType')->by_categories($oxfordshire->id, @contacts);
```

Comment:
Is it not an area ID that is passed to this, not a body ID? The area ID is 2237, that is fine.